### ISSUE TITLE
[Template] How these two parameters affect batch processing?

### ISSUE THREAD

**Issue by mguidoti** [gsautter/goldengate-imagine#839]
Hi Guido,

Can you help me understand how GGI behaves on this issue?

There are two parameters that works differently but address similar issues. One is the `Maximum pages after first` on the Anchor settings; the other is `Number of Cover Pages`, on General Page Layout. I think the first applies only to the anchor, and the second simply deletes (or doesn't load) the x amount of pages, from start to finish, when converting the PDF to IMF.

Right?

Now, how does this impact batch processing? Meaning that, if I set the first to 2, will the 2 initial pages be ignored for everything else or just for the anchor? And what if I do it in the second? 

In the case of the four MNHN journals, like Zoosystema, where we do have 2 initial cover pages, should I use the second parameter instead?

Thanks,

**Comment by gsautter** [comment]
You are right in that these two parameters address similar issues, but they don't address the same issue:
- `Maximum pages after first` in anchors defines how far down a document to look for an anchor match
- `Number of Cover Pages` defines how many cover pages to expect _at most_, mainly to ignore them when seeking page images in scans (because these meta pages are usually born-digital even for otherwise scanned documents)

Basically, the former has to be available when _determining_ which template to use, at which point the latter is not available. Also, even without any meta pages an anchor can be a few pages down the document, e.g. with books.

On the architecture level, `Maximum pages after first` is used in the anchor mechanism, which is specific to a particular style provider implementation, while `Number of Cover Pages` is specific to the PDF decoder, which is only one of many consumers of template parameters. That said, having both also to some degree is the cost of keeping our system modular, which facilitates using the very same template mechanism e.g. for XML documents (with a different provider and an XML specific anchor mechanism), as well as using the PDF decoder without any anchors, e.g. with a template specified by the user.

Hope that clarifies matters a bit.

**Comment by mguidoti** [comment]
Yes, it did. I wasn't too off, then. Good to know!

Thanks, Guido.
